packages/@lwc/ssr-compiler/src/compile-template/index.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { generate } from 'astring';
+import { is, builders as b } from 'estree-toolkit';
+import { parse } from '@lwc/template-compiler';
+import { esTemplate } from '../estemplate';
+import { templateIrToEsTree } from './ir-to-es';
+import { optimizeAdjacentYieldStmts } from './shared';
+
+import type {
+    Node as EsNode,
+    Statement as EsStatement,
+    Literal as EsLiteral,
+    ExportDefaultDeclaration as EsExportDefaultDeclaration,
+} from 'estree';
+
+const isBool = (node: EsNode | null) => is.literal(node) && typeof node.value === 'boolean';
+
+const bExportTemplate = esTemplate<
+    EsExportDefaultDeclaration,
+    [EsLiteral, EsStatement[], EsLiteral]
+>`
+    export default async function* tmpl(props, attrs, slotted, Cmp, instance, stylesheets) {
+        if (!${isBool} && Cmp.renderMode !== 'light') {
+            yield \`<template shadowrootmode="open"\${Cmp.delegatesFocus ? ' shadowrootdelegatesfocus' : ''}>\`
+        }
+
+        for (const stylesheet of stylesheets ?? []) {
+            // TODO
+            const token = null;
+            const useActualHostSelector = true;
+            const useNativeDirPseudoclass = null;
+            yield '<style type="text/css">';
+            yield stylesheet(token, useActualHostSelector, useNativeDirPseudoclass);
+            yield '</style>';
+        }
```

Comment:
This would also require implementing scoped styles.

Reply:
Yes, I have a work item to implement scoped styles after this PR is merged.